src/entity.js

```diff
@@ -677,101 +693,113 @@ Entity.setup = function(options) {
-           "Expected options.account to be a string, or undefined");
-    // Create auth client to fetch SAS from auth.taskcluster.net
-    var auth = new taskcluster.Auth({
-      credentials:    options.credentials,
-      baseUrl:        options.authBaseUrl
-    });
-    // Create azure table client with logic for fetch SAS
-    client = new azure.Table({
-      timeout:          AZURE_TABLE_TIMEOUT,
-      agent:            options.agent,
-      accountId:        options.account,
-      minSASAuthExpiry: options.minSASAuthExpiry,
-      sas: function() {
-        return auth.azureTableSAS(
-          options.account,
-          options.table
-        ).then(function(result) {
-          return result.sas;
-        });
-      }
-    });
-  } else {
-    // Create client using credentials already present
-    assert(options.credentials.accountName, "Missing accountName");
-    assert(options.credentials.accountKey ||
-           options.credentials.sas,         "Missing accountKey or sas");
-    // Create azure table client with accessKey
-    client = new azure.Table({
-      timeout:      AZURE_TABLE_TIMEOUT,
-      agent:        options.agent,
-      accountId:    options.credentials.accountName,
-      accessKey:    options.credentials.accountKey,
-      sas:          options.credentials.sas
-    });
-  }
-
-  // Store reference to azure table client
-  subClass.prototype.__client = client;
-
   // Reporter for statistics
   var reporter = function() {};
   if (options.drain) {
     reporter = series.AzureTableOperations.reporter(options.drain);
   }
 
-  // Create table client wrapper, to record statistics and bind table name
-  subClass.prototype.__aux = {};
-  [
-    'createTable',
-    'deleteTable',
-    'getEntity',
-    'queryEntities',
-    'insertEntity',
-    'updateEntity',
-    'deleteEntity'
-  ].forEach(function(name) {
-    // Bind table name
-    var method = client[name].bind(client, options.table);
-
-    // Record statistics
-    subClass.prototype.__aux[name] = function() {
-      var start = process.hrtime();
-      return method.apply(client, arguments).then(function(result) {
-        var d = process.hrtime(start);
-        reporter({
-          component:    options.component,
-          process:      options.process,
-          duration:     d[0] * 1000 + (d[1] / 1000000),
-          table:        options.table,
-          method:       name,
-          error:        'false'
-        });
-        return result;
-      }, function(err) {
-        var d = process.hrtime(start);
-        reporter({
-          component:    options.component,
-          process:      options.process,
-          duration:     d[0] * 1000 + (d[1] / 1000000),
-          table:        options.table,
-          method:       name,
-          error:        (err ? err.code : null) || 'UnknownError'
-        });
-        throw err;
+  if (options.inMemory) {
+    if (!inmemory) {
+      inmemory = require('./inmemory'); // lazy-loaded
```

Comment:
There is no need to make this a global reference... `require` won't reload, and it's super expensive...
Would be worth it if we were referencing `inmemory` elsewhere, but we aren't...